pypdf/generic/_base.py

```diff
@@ -619,6 +619,15 @@ def write_to_stream(
         stream.write(binascii.hexlify(self))
         stream.write(b">")
 
+    def __str__(self) -> str:
+        charset_to_try = ["utf-16"] + list(NameObject.CHARSETS)
+        for enc in charset_to_try:
+            try:
+                return self.decode(enc)
+            except UnicodeDecodeError:
+                pass
+        return ""
```

Comment:
Shouldn't this rather raise an exception? For now, I consider it misleading to return an empty string for non-empty inputs.

Reply:
Again, the calling code (`DocumentInformation.title`) assumes that `_get_text` will return `''` if failed to decode, and I didn't want to change any unrelated behavior other than the bug I am fixing. But it does make sense, so I will change and hopefully no test will fail.